src/views/dasboard/index.js

```diff
@@ -0,0 +1,139 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import classnames from 'classnames'
+import { useDispatch, useSelector } from 'react-redux'
+
+import Header from '_components/header'
+import { CHANGE_NAME, LOGGED_AS, USER_ID } from '_utils/dashboard'
+import { getUserSelector } from '_modules/user/selectors'
+import Input from '_components/input'
+import Button, { ButtonTheme } from '_components/button'
+import { getUser, updateUser } from '_modules/user/actions'
+import Spinner, { SpinnerTheme } from '_components/spinner'
+
+import styles from './styles.css'
+
+const ALLOW_IMAGES_EXTENSIONS = ['.jpeg', '.jpg', '.png']
+
+function getBase64(file) {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader()
+    reader.readAsDataURL(file)
+    reader.onload = () => resolve(reader.result)
+    reader.onerror = error => reject(error)
+  })
+}
+
+const Dashboard = () => {
+  const user = useSelector(getUserSelector)
+  const [userNewName, setUserNewName] = useState('')
+  const [isLoading, setIsLoading] = useState(false)
+
+  const dispatch = useDispatch()
+
+  const handlePictureChange = useCallback(
+    event => {
+      const file = event.target.files[0]
+      getBase64(file).then(data => {
+        dispatch(updateUser({ avatar: data }, user.id))
+      })
+    },
+    [dispatch, user.id]
+  )
+
+  const handleChangeAvatar = useCallback(() => {
+    const inputFile = document.createElement('input')
+    inputFile.value = ''
+    inputFile.type = 'file'
+    inputFile.accept = ALLOW_IMAGES_EXTENSIONS
+    inputFile.onchange = handlePictureChange
+    inputFile.click()
+  }, [handlePictureChange])
+
+  const handleOnChange = (event, state) => {
+    const { value } = event.target
+
+    state(value)
+  }
+
+  const handleSubmit = async event => {
+    event.preventDefault()
+    setIsLoading(true)
+
+    if (userNewName) {
+      await dispatch(updateUser({ name: userNewName }, user.id))
+    }
+    setIsLoading(false)
+  }
+
+  useEffect(() => {
+    setIsLoading(true)
+    dispatch(getUser())
+    setIsLoading(false)
+  }, [dispatch])
```

Comment:
this setIsLoading makes no sense.
You're not using any promises or async functions here, so `setIsLoading(false)` is not gonna wait for `dispatch(getUser())` to finish to be called. This does not ensures that your loading state here is actually the loading state of getting the user. It should be either like what you did on handleSubmit, or you should deal with it directly on the reducer (with pending/fulfilled/reject states)